sql/sql_table.cc

```diff
@@ -4203,6 +4203,20 @@ mysql_prepare_create_table(THD *thd, HA_CREATE_INFO *create_info,
             DBUG_RETURN(TRUE);
           }
         }
+
+        /* Check that there's no repeating foreign key constraint names. */
+        List_iterator_fast<Key> key_it(alter_info->key_list);
+        Key *key;
+        while ((key= key_it++))
```

Comment:
I do not see a check against duplicated `FOREIGN KEY` constraint names. I do see a check against the same name being used for both `CHECK` and `FOREIGN KEY` constraints. Can you revise the comment?
I would suggest introducing `const` and reducing the scope of a variable, like C++98 allows us to do:
```c++
while (const Key* key= key_it++)
```